src/query/query_executor.c

```diff
@@ -20591,14 +20711,29 @@ qexec_execute_build_indexes (THREAD_ENTRY * thread_p, XASL_NODE * xasl, XASL_STA
 
   for (i = 0, attrepr = rep->attributes; i < rep->n_attributes; i++, attrepr++)
     {
-      attr_name = or_get_attrname (&class_record, attrepr->id);
+      string = NULL;
+      alloced_string = 0;
+
+      error = or_get_attrname (&class_record, attrepr->id, &string, &alloced_string);
+      if (error != NO_ERROR)
+	{
+	  ASSERT_ERROR ();
+	  GOTO_EXIT_ON_ERROR;
+	}
+
+      attr_name = string;
       if (attr_name == NULL)
 	{
 	  continue;
 	}
 
       attr_names[i] = (char *) attr_name;
       attr_ids[i] = attrepr->id;
+
+      if (string != NULL && alloced_string == 1)
+	{
+	  db_private_free_and_init (thread_p, string);
+	}
```

Comment:
@lucianlazar2211 

Please notice that `attr_name[i]` may point to `string` which was alloced for decompression.